apps/storefront/src/pages/Login/index.tsx

```diff
@@ -30,31 +30,54 @@ import { getCurrentCustomerInfo } from '@/utils/loginInfo';
 import { type PageProps } from '../PageProps';
 
 import LoginWidget from './component/LoginWidget';
-import { loginCheckout, LoginConfig, LoginInfoInit } from './config';
+import { loginCheckout, LoginConfig } from './config';
 import LoginForm from './LoginForm';
 import LoginPanel from './LoginPanel';
 import { LoginContainer, LoginImage } from './styled';
 
-const initialLoginInfo = {
-  loginTitle: 'Sign In',
-  loginBtn: 'Sign in',
-  createAccountPanelTittle: 'Create Account Panel Title',
-  CreateAccountButtonText: 'Create Account',
-  btnColor: 'red',
-  widgetBodyText: '',
-  displayStoreLogo: false,
+type AlertColor = 'success' | 'info' | 'warning' | 'error';
+
+const useMasquerade = () => {
+  const isAgenting = useAppSelector(({ b2bFeatures }) => b2bFeatures.masqueradeCompany.isAgenting);
+  const b2bId = useAppSelector(({ company }) => company.customer.b2bId);
+  const salesRepCompanyId = useAppSelector(({ b2bFeatures }) => b2bFeatures.masqueradeCompany.id);
+  const storeDispatch = useAppDispatch();
+
+  const isMasquerade = isAgenting && typeof b2bId === 'number';
+
+  const endMasquerade = useCallback(async () => {
```

Comment:
here's the fix for the loading not happening

Reply:
Ah, of course. Thanks for the fix.